lsh/indexing/HashFunction.java

```diff
@@ -33,14 +33,14 @@ public HashFunction(int numberOfDimensions, double slotWidthW, Random randomNumb
 		
 	}
 
-	public HashFunction(int numberOfDimensions, double slotWidthW, double eigenVector[]) {
+	public HashFunction(int numberOfDimensions, double slotWidthW, double eigenVector[], Random randomNumberGenerator) {
 		this.numberOfDimensions = numberOfDimensions;
-		this.hashFunctionCoefficients = new ArrayList<Double>(numberOfDimensions);
+		this.hashFunctionCoefficients = new ArrayList<Double>();
 
 		for (int i = 0; i < numberOfDimensions; i++)
 			hashFunctionCoefficients.add(eigenVector[i]);
 
-		this.offset = 0.0;
+		this.offset = randomNumberGenerator.nextDouble() * slotWidthW;
```

Comment:
wont this offset then be normalized by slot width and wind up being the addition of just a small double? the paper mentions real offsets between 0 and w. technically a small double would be fine but on a glance it seems odd